app/presenters/tree_builder_automate_simulation_results.rb

```diff
@@ -0,0 +1,78 @@
+class TreeBuilderAutomateSimulationResults < TreeBuilder
+  has_kids_for Hash, [:x_get_tree_hash_kids]
+  def initialize(name, type, sandbox, build = true, root = nil)
+    @root = root
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def tree_init_options(_tree_name)
+    {
+      :full_ids => true,
+      :add_root => false,
+      :expand   => true,
+      :lazy     => false
+    }
+  end
+
+  def set_locals_for_render
+    locals = super
+    locals.merge!(:id_prefix     => "aesimulation_",
+                  :autoload      => false,
+                  :cfme_no_click => true,
+                  :onclick       => false)
+  end
+
+  def root_options
+    []
+  end
+
+  def x_get_tree_roots(count_only = false, _options)
+    objects = []
+    xml = MiqXml.load(@root).root
+    xml.each_element do |el|
+      objects.push(get_root_elements(el, xml.index(el)))
+    end
+    objects
+  end
```

Comment:
@martinpovolny i tried ``` .each_element.with_object ``` but that didn't seem to work on xml, the only way i could get this to work was using each_element and then finding index of element to get it's index to build uniq node id.